### ISSUE TITLE
[ERROR] Malformed game path

### ISSUE THREAD

**Issue by arthurmohn** [aclist/dztui#34]
**Info**
DZGUI version: 1.0
Mode (normal/debug): normal
Server ID:
Linux distribution: fedora 36

**Bug description**
i have this error when opening the DZGUI through Steam: "[ERROR] Malformed game path" 
After pressing OK: "Malformed config file. This is probably user error.Start first-time setup process again?"
and after pressing Yes the app justs stops running.

![image](https://user-images.githubusercontent.com/52223014/193432347-aeb11bef-7ad2-4a27-b32e-5eaa01527bae.png)

![image](https://user-images.githubusercontent.com/52223014/193432349-7106be19-2b00-4198-b32b-5737c1b4b2e4.png)

**To Reproduce**
1. Add game to Steam
2. Click on 'Play'

**Expected behavior**
I expected the app to open like normal. i can open DZGUI from the app launcher.

**Comment by aclist** [comment]
I was unable to reproduce this after deliberately malforming the config file and starting the application again. Clicking yes should kick off the new config creation process.

I'm wondering about how you added the game to Steam and whether it is spawning processes in a subshell correctly. Which file did you add to Steam? Did you add a .sh file or a .desktop file?

Also, can you invoke the following:
1. Select Toggle debug mode from main menu
2. Go to Debug options at bottom of list
3. Select Generate debug log
4. Copy the content of the log (it will tell you where it was saved to) and paste it here

You stated that you are using DZGUI 1.0. The current version is 2.6.1. Can you check the version number at the top of the application?

**Comment by arthurmohn** [comment]
Sorry, It is the 2.6.1 version.
I added the .desktop on the path: '/home/arthur/.local/share/applications/'.

I could only switch the app to debug mode by opening it from the app laucher. This is the log content:
````
Version: 2.6.1
Branch: stable
Whitelist: 16656417
Path: 
Linux: Linux 5.19.12-200.fc36.x86_64 x86_64

Mods:
````

**Comment by aclist** [comment]
OK, and what is the `steam_path` value in the file `/home/arthur/.config/dztui/dztuirc`? Empty?

And what is the result of this command?
`find / -type d \( -path "/proc" -o -path "*/timeshift" -o -path "/tmp" -o -path "/usr" -o -path "/boot" -o -path "/proc" -o -path "/root" -o -path "/run" -o -path "/sys" -o -path "/etc" -o -path "/var" -o -path "/run" -o -path "/lost+found" \) -prune -o -regex ".*/steamapps/common/DayZ$" -print -quit 2>/dev/null`

**Comment by arthurmohn** [comment]
I just found the solution. My steam library is in another drive and this drive was mounted in a wierd location. I just mounted it in the  corrected place and it worked. Thanks for the help.

**Comment by aclist** [comment]
OK, I thought that might be the case. I will try to investigate why the "first-time setup" message failed, though.

**Comment by aclist** [comment]
Thanks for your patience. I did a major overhaul/simplification of the way first-time setup works and it should be nearly impossible for a user to break their config now. It should also be faster for the app to discover where Steam/DayZ is installed.

I also investigated the error you mentioned about the application crashing when you click OK to start the first-time setup again. I found and fixed that bug, too. There was an initial process that was getting pinned to this menu and was blocking the rest of the app from launching. When the menu despawned, it was killing the rest of the app with it.

I'll leave this open for a few days and close it after it is out in the wild a bit.
